lib/wallet/api/http/Cardano/Wallet/Api/Types.hs

```diff
@@ -1206,6 +1210,80 @@ newtype ApiEncryptMetadata = ApiEncryptMetadata
     { passphrase :: ApiT (Passphrase "lenient") }
     deriving (Eq, Generic, Show)
     deriving (FromJSON, ToJSON) via DefaultRecord ApiEncryptMetadata
+
+newtype DRepKeyHash = DRepKeyHash { getDRepKeyHash :: ByteString }
+    deriving (Generic, Eq, Ord)
+
+instance NFData DRepKeyHash
+
+instance Show DRepKeyHash where
+    show p = "(DRep key hash " <> show (encodeDRepKeyHashBech32 p) <> ")"
+
+-- | Encode 'DRepKeyHash' as Bech32 with "drep_vkh" hrp.
+encodeDRepKeyHashBech32 :: DRepKeyHash -> T.Text
+encodeDRepKeyHashBech32 =
+    Bech32.encodeLenient hrp
+        . Bech32.dataPartFromBytes
+        . getDRepKeyHash
+  where
+    hrp = [Bech32.humanReadablePart|drep_vkh|]
+
+-- | Decode a Bech32 encoded 'DRepKeyHash'.
+decodeDRepKeyHashBech32 :: T.Text -> Either TextDecodingError DRepKeyHash
+decodeDRepKeyHashBech32 t =
+    case fmap Bech32.dataPartToBytes <$> Bech32.decodeLenient t of
+        Left _ -> Left textDecodingError
+        Right (_, Just bytes) ->
+            Right $ DRepKeyHash bytes
+        Right _ -> Left textDecodingError
+      where
+        textDecodingError = TextDecodingError $ unwords
+            [ "Invalid DRep key hash: expecting a Bech32 encoded value"
+            , "with human readable part of 'drep_vkh'."
+            ]
+
+newtype DRepScriptHash = DRepScriptHash { getDRepScriptHash :: ByteString }
+    deriving (Generic, Eq, Ord)
+
+instance NFData DRepScriptHash
+
+instance Show DRepScriptHash where
+    show p = "(DRep script hash " <> show (encodeDRepScriptHashBech32 p) <> ")"
+
+-- | Encode 'DRepScriptHash' as Bech32 with "drep_script" hrp.
+encodeDRepScriptHashBech32 :: DRepScriptHash -> T.Text
+encodeDRepScriptHashBech32 =
+    Bech32.encodeLenient hrp
+        . Bech32.dataPartFromBytes
+        . getDRepScriptHash
+  where
+    hrp = [Bech32.humanReadablePart|drep_script|]
+
+-- | Decode a Bech32 encoded 'DRepScriptHash'.
+decodeDRepScriptHashBech32 :: T.Text -> Either TextDecodingError DRepScriptHash
+decodeDRepScriptHashBech32 t =
+    case fmap Bech32.dataPartToBytes <$> Bech32.decodeLenient t of
+        Left _ -> Left textDecodingError
+        Right (_, Just bytes) ->
+            Right $ DRepScriptHash bytes
+        Right _ -> Left textDecodingError
+      where
+        textDecodingError = TextDecodingError $ unwords
+            [ "Invalid DRep Script hash: expecting a Bech32 encoded value"
+            , "with human readable part of 'drep_script'."
+            ]
```

Comment:
```suggestion
decodeDRepScriptHashBech32 :: T.Text -> Either TextDecodingError DRepScriptHash
decodeDRepScriptHashBech32 t =
    case fmap Bech32.dataPartToBytes <$> Bech32.decodeLenient t of
        Left _ -> Left textDecodingError
        Right (hrp', Just bytes)
            | hrp' == hrp -> Right $ DRepScriptHash bytes
        Right _ -> Left textDecodingError
      where
        textDecodingError = TextDecodingError $ unwords
            [ "Invalid DRep Script hash: expecting a Bech32 encoded value"
            , "with human readable part of 'drep_script'."
            ]
        hrp = [Bech32.humanReadablePart|drep_script|]
```

I would recommend to also check the human readable part — the error message suggests that it should be present anyway.

Reply:
good idea. done